api/provider/linkedin.go

```diff
@@ -127,17 +127,22 @@ func (g linkedinProvider) GetUserData(ctx context.Context, tok *oauth2.Token) (*
 		})
 	}
 
+	var avatarURL string
+	if len(u.AvatarURL.DisplayImage.Elements) > 0 {
```

Comment:
How about we move this logic as a function on the `linkedInUser` object, since I think this needs a deeper length check. 
```suggestion
	if len(u.AvatarURL.DisplayImage.Elements) > 0 && len(u.AvatarURL.DisplayImage.Elements[0].Identifiers) > 0 {
```